test/metabase/query_processor_test/explicit_joins_test.clj

```diff
@@ -772,12 +772,20 @@
                         "Products Renamed → Rating"
                         "Products Renamed → Created At"]
                        (map :display_name (get-in results [:data :results_metadata :columns])))))
-              (is (= [[6 1 60 29.8 1.64 31.44 nil "2019-11-06T16:38:50.134Z" 3 2
-                       60 "4819782507258" "Rustic Paper Car" "Doohickey" "Stroman-Carroll" 19.87 4.1 "2017-12-16T11:14:43.264Z"]
-                      [10 1 6 97.44 5.36 102.8 nil "2020-01-17T01:44:37.233Z" 2 2
-                       6 "2293343551454" "Small Marble Hat" "Doohickey" "Nolan-Wolff" 64.96 3.8 "2017-03-29T05:43:40.15Z"]]
-                     (mt/formatted-rows [int int int 2.0 2.0 2.0 2.0 str int int
-                                         int str str str str 2.0 2.0 str]
+              (is (= (if (= driver/*driver* :mongo)
```

Comment:
The order of columns is different in Mongo. What's the best fix?

Reply:
Fix the order of the columns are returned by MongoDB? 

The order columns get returned in by MBQL is pretty unambiguous and I'm worried if this is broken for Mongo it's going to break other stuff without us realizing. There is some stuff like `metabase.query-processor.middleware.annotate/mbql-cols` or `add-column-info` that takes a fully-preprocessed MBQL query and can determine the exact shape of the results and returns metadata based on that

Reply:
The _real_ fix would require a change in monger, but I'll have a go at fixing it for top level fields.